### PR TITLE
Add notes to contributors guide

### PR DESCRIPTION
**Pull request by jbcden**
This adds some notes around some issues I ran into when trying to run the tests for Diesel.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -109,6 +113,13 @@ Thank you! We'll try to respond as quickly as possible.
    by executing `bin/test`. (Initially, this will take a while to compile
    everything.)
 
+   - One thing to note is that this script optionally runs Clippy checks as well.
+   Since this requires the nightly compiler these checks are disabled by default.
+   In order to run these checks you can do:
+   ```bash
+   bin/test -- --with-clippy
```
Comment (weiznich) [inline CONTRIBUTING.md]: Change this to something like `--disable-clippy` as Sean mentioned in the gitter channel.

**Comment by jbcden** [comment]
@weiznich I just saw your comment, how do you feel about `--no-clippy`? I also wasn't sure if 1acc79212ff4647ee677b7ac7bc829515843ab30 should be pulled into a separate PR or if it was ok to lump it into this one. Thoughts?

File: bin/test [hunk 2]
```diff
@@ -19,6 +26,8 @@ elif [ "$1" = "compile" ]; then
   shift
   (cd diesel_compile_tests && cargo test $*)
 else
+  echo "TEST"
```
Comment (weiznich) [inline bin/test]: This two lines should be removed.

**Comment by weiznich** [comment]
> I just saw your comment, how do you feel about --no-clippy?

`--no-clippy` is also fine in my opinion

> I also wasn't sure if 1acc792 should be pulled into a separate PR or if it was ok to lump it into this one. Thoughts?

I don't think this needs a separate pull request.

**Comment by jbcden** [comment]
> This two lines should be removed.

🤦‍♂️ removed in df38cf5c8fe88e238e59a9c2ed712891d74f5820

File: bin/test [hunk 3]
```diff
@@ -1,7 +1,14 @@
 #!/usr/bin/env bash
 set -e
 
-CLIPPY="lint"
+echo $*
```
Comment (Eijebong) [inline bin/test]: This needs to be removed too ^_^

**Comment by jbcden** [comment]
> This needs to be removed too ^_^

🤦‍♂️ I think I got all the `echo`s this time.

**Pull request merged**